app/src/main/java/com/duckduckgo/app/notification/model/PrivacyProtectionNotification.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.notification.model
+
+import android.content.Context
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.notification.NotificationHandlerService.NotificationEvent.APP_LAUNCH
+import com.duckduckgo.app.notification.NotificationHandlerService.NotificationEvent.CANCEL
+import com.duckduckgo.app.notification.NotificationRegistrar
+import com.duckduckgo.app.notification.db.NotificationDao
+import com.duckduckgo.app.privacy.db.PrivacyProtectionCountDao
+import kotlinx.coroutines.runBlocking
+
+
+class PrivacyProtectionNotification(
+    private val context: Context,
+    private val notificationDao: NotificationDao,
+    private val privacyProtectionCountDao: PrivacyProtectionCountDao
+) : SchedulableNotification {
+
+    override val id = "com.duckduckgo.privacy.privacyprotection"
+
+    override val specification
+        get() = PrivacyProtectionNotificationSpecification(context, privacyProtectionCountDao)
+
+    override val launchIntent: String = APP_LAUNCH
+
+    override val cancelIntent: String = CANCEL
+
+    override suspend fun canShow(): Boolean {
+        return !notificationDao.exists(id)
+    }
+}
+
+class PrivacyProtectionNotificationSpecification(
+    context: Context,
+    private val privacyProtectionCountDao: PrivacyProtectionCountDao
+) : NotificationSpec {
+
+    override val channel = NotificationRegistrar.ChannelType.TUTORIALS
+    override val systemId = NotificationRegistrar.NotificationId.PrivacyProtection
+    override val name = "Privacy protection"
+    override val icon = R.drawable.notification_sheild_lock
+    override val launchButton: String = context.getString(R.string.privacyProtectionNotificationLaunchButton)
+
+    private val trackers = runBlocking { privacyProtectionCountDao.getTrackersBlockedCount().toInt() }
```

Comment:
I worry this disguises that this call is doing real work whenever it is created. It's not obvious when you instantiate this class it'll go retrieve from the DB. 

For example, i'm very surprised to see that this line invokes a DB call

`val specification = notification.specification`

Reply:
I've turned it into a suspend function called `buildSpecification` to make the fact that it's doing heavy lifting more apparent.

Reply:
Nice change; I think that's much clearer 👌